src/tabs.js

```diff
@@ -0,0 +1,209 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA ;
+ */
+/**
+ * @author Martin Nicholson <martin@taotesting.com>
+ */
+/**
+ * @example
+ * var instance = tabs({
+ *   renderTo: $container,
+ *   tabs: [
+ *     { label: 'TAO Local', name: 'local-delivery' },
+ *     { label: 'TAO Remote', name: 'remote-delivery' },
+ *     { label: 'LTI-based', name: 'lti-delivery', disabled: true }
+ *   ],
+ *   activeTabIndex: 1
+ * });
+ *
+ */
+import $ from 'jquery';
+import _ from 'lodash';
+import component from 'ui/component';
+import tabsTpl from 'ui/tabs/tpl/tabs';
+import 'ui/tabs/css/tabs.css';
+
+const ns = 'tabs';
+
+/**
+ * Default config
+ * @param {Integer} activeTabIndex - the index of the tab to start on
+ * @param {Boolean} showHideTargets - if true, no need to wire up tabs to panels, it's automatic
+ * @param {String} targetDataAttr - the data attribute suffix used to connect tabs to panels
+ */
+const tabsDefaults = {
+    activeTabIndex: 0,
+    showHideTargets: true,
+    targetDataAttr: 'panel'
+};
+
+/**
+ * In-memory data
+ */
+let tabs = [];
+
+/**
+ * API of the tabs component
+ * @exports ui/tabs
+ */
+const tabsApi = {
+    /**
+     * Set new values for the tabs
+     * @param {Array} newTabs
+     * @returns {tabs} instance
+     * @throws {TypeError} on non-Array tabs
+     */
+    setTabs(newTabs) {
+        if (!_.isArray(newTabs)) {
+            throw new TypeError('The provided tabs are not a valid array');
+        }
+        tabs = [...newTabs];
+        return this;
+    },
+
+    /**
+     * Retrieve internal tabs array
+     * @returns {Array} tabs list
+     */
+    getTabs() {
+        return tabs;
+    },
+
+    /**
+     * Wires up the onClick functions of the received tabs
+     * @returns {tabs} instance
+     */
+    connectTabs() {
+        const $tabBar = this.getElement();
+        const attr = this.config.targetDataAttr;
+
+        for (let tab of tabs) {
+            $tabBar.find(`[data-controlled-${attr}="${attr}-${tab.name}"]`)
+                .off('click')
+                .on('click', () => {
+                    this.activateTabByName(tab.name);
+                });
+        }
+        return this;
+    },
+
+
+    /**
+     * Activates a single tab by its name
+     * (pass-through method to activateTabByIndex)
+     * @param {String} name - human-readable identifier
+     * @param {Boolean} [callOnClick=true] - if false, skips the onClick call
+     * @returns {tabs} instance
```

Comment:
returns with the instance or nothing :)

Reply:
You mean if it throws? I don't think that should be documented as undefined return type.